internal/overlord/servstate/manager.go

```diff
@@ -12,7 +12,21 @@ import (
 	"github.com/canonical/pebble/internal/servicelog"
 )
 
-type ServiceManager struct {
+type ServiceManager interface {
```

Comment:
Curious why you changed this to an interface? Normally in Go packages return concrete types, and interfaces are only created where needed (e.g., in `Overlord` itself, so the `ServiceManager` can be mocked out).